1.3.24 update from Ledger with Veil support

Kryptoxic opened the pull request:
Updated the app to 1.3.16 and also included Veil support.

## Bounty
Bounty address: bv1ql9t8rzaqqtet5urz2j4ywqn5e38efzxr7yvemc

Basecoin address is due to using an exchange address. I'm currently getting funds together for a new PC because my previous computer died on me so I'm currently slow in pushing out codes. Help me out with a donation to a new PC! :D

btchip committed:
Initial commit

Commit:
Initial import

Commit:
Nano S release

Build for Blue SDK 1.2 / Nano S 1.0 release
Nano S UI
Enable message signing for arbitrary message length
Reenable original P2SH shortcut while waiting for multiple output support

Commit:
Add release signature for nanos-10

Commit:
Add apilevel for new ledgerblue loader API

Commit:
Cleanup, adjust derivation API to API level 5, adjust linker script and NVRAM location

Commit:
Add FLAG_DERIVE_MASTER for newer firmware / GreenAddress support

Commit:
Multicoin support - support specific dashboard icon, fix coin ID in message signature, implement GET COIN VERSION / SET ALTERNATE COIN VERSION

Multicoin support

Commit:
Fix change output detection that got lost in HW.1 conversion

Commit:
Sanity check - do not authorize multiple similar inputs from different trust origins in the same transaction - also lost in HW.1 conversion. Caught by N.Dorier

Commit:
Update Makefile for release and multicoin support

Commit:
Add icons

Commit:
Add support for two byte versions (Zcash)

Commit:
Fix display when no decimals are present

Commit:
Add U2F proxy

Commit:
Add U2F proxy client API

Commit:
Missing U2F configuration

Commit:
Fix external badge icon display

Commit:
Fix parsing of Zcash transactions v2 with joinsplit

Commit:
Add U2F proxy client API for P2SH inputs

Commit:
Fix input loop

Commit:
Refresh icons for Blue 2.0

Commit:
Peercoin support

Commit:
U2F proxy optimization, Blue 2.0 UI

Commit:
Merge branch 'master' of https://github.com/LedgerHQ/blue-app-btc

btchip committed:
Display public key, support stremed outputs, support P2SH parsing (Nano-S only for the time being), add Peercoin build

btchip committed:
Build with new SDK scripts, fix USB disconnect, add pending icons

btchip committed:
Display the full (scrollable) destination address on Nano S
Fix screen lock management for large TXs - only require an unlocked screen for the firs$
Reorganize Makefile
Review UI for Blue SDK and upcoming firmware 1.4.0
Add QR Code for Blue (pending SDK update)
Add POSW

btchip committed:
Fix mixed Segwit/regular inputs signing, add Bitcoin Cash app

btchip committed:
Add native Segwit addresses support and pivx, vertcoin, viacoin

btchip committed:
Add Stealthcoin, fix Peercoin based altcoins Satoshi unit

btchip committed:
Add Bitcoin Gold support

btchip committed:
Fix Bitcoin Gold coin profile

btchip committed:
Fix Bitcoin Gold Segwit support

btchip committed:
Add DGB, Hcash, QTUM

btchip committed:
Fix native Segwit address

btchip committed:
Update to Nano S SDK 1.4.1

btchip committed:
Fix native Segwit

Raulo committed:
Fixes #21. Incorrect display of bech32

btchip committed:
Merge pull request #27 from Raulo/master

Fixes #21. Incorrect display of bech32 P2WPKH

btchip committed:
Add double interface, modify Trusted Input format to avoid using a cryptographic export

Gbillou committed:
Rely on BOLOS_ENV if defined to find CLANG/GCCPATHs

btchip committed:
Merge pull request #31 from Gbillou/master

Rely on BOLOS_ENV if defined to find CLANG/GCCPATHs

btchip committed:
Add Bitcoin Private, ZenCash, CashAddr support

josephnicholas committed:
Script Fixes for P2SH Transaction

pa-t committed:
adds comments

btchip committed:
Merge pull request #38 from ZencashOfficial/master

ZenCash P2SH TX Support!

btchip committed:
Update Peercoin Coin ID

Commit:
update stealthcoin to stealth

btchip committed:
Updates for Blue build

btchip committed:
Add missing Blue icons

btchip committed:
Merge branch 'master' into update_stealthcoin_name

btchip committed:
Merge pull request #46 from TamtamHero/update_stealthcoin_name

Updated name of stealthcoin to stealth

btchip committed:
Add Zcoin

btchip committed:
Add Litecoin Bech32 prefix

TamtamHero committed:
update address prefixes for DigiByte

TamtamHero committed:
Rebrand Horizen

TamtamHero committed:
updated Makefile for auto-deploy

TamtamHero committed:
fix horizen icon

btchip committed:
Merge pull request #53 from TamtamHero/horizen_rebranding

Horizen rebranding & auto-deploy

btchip committed:
Merge pull request #52 from TamtamHero/master

update address prefixes for DigiByte

oto-ledger committed:
Update Makefile

Take into account dependency version to avoid confusion

btchip committed:
Merge pull request #58 from oto-ledger/patch-2

Update Makefile

samadsajanlal committed:
add GameCredits support

samadsajanlal committed:
GameCredits - add to ListVariants and fix coin_kind

TamtamHero committed:
added get_coin_version apdu to doc

samadsajanlal committed:
GameCredits: update with approved graphics

btchip committed:
Merge pull request #67 from samadsajanlal/master

add GameCredits support

btchip committed:
Merge pull request #73 from TamtamHero/master

adding get_coin_version apdu to doc

TamtamHero committed:
Bump version 1.2.10

btchip committed:
Merge pull request #74 from TamtamHero/patch-1

Bump version 1.2.10

TamtamHero committed:
add a privacy preserving feature for public key exports

TamtamHero committed:
update doc for privacy feature

jibeee committed:
Fixes for base 58 encoding and decoding

TamtamHero committed:
Make privacy protection effective only when communication is done through U2F

btchip committed:
Merge pull request #75 from TamtamHero/pubkey_privacy

Add a privacy preserving feature for public key exports

btchip committed:
Parse Omni Simple Send, bump version, remove temporarily ZCash variant (using the branch for Blake2b)

jibeee committed:
Base58: make function prototypes consistent

btchip committed:
Revert "Add a privacy preserving feature for public key exports"

btchip committed:
Merge pull request #79 from LedgerHQ/revert-75-pubkey_privacy

Revert "Add a privacy preserving feature for public key exports"

btchip committed:
Merge pull request #76 from jibeee/master

    Fixes for base 58 encoding and decoding

TamtamHero committed:
Fix missing Zcoin Blue icon

TamtamHero committed:
Add public key protection for U2F transport

btchip committed:
Merge pull request #82 from TamtamHero/privacy_feature

Add a privacy preserving feature for public key exports, V2

TamtamHero committed:
re-enable OMNI

btchip committed:
Merge pull request #83 from TamtamHero/master

Re-enable Omni parsing

btchip committed:
Revert #76 temporarily

TamtamHero committed:
Remove unused branches on some display checks

jibeee committed:
Base58: code was leaving the encoding loop too early in some cases

TamtamHero committed:
Remove unused if branch on some display checks

TamtamHero committed:
Update privacy feature wording

btchip committed:
Merge pull request #87 from TamtamHero/master

Wording update on privacy feature + cleaning

Commit:
Add ZClassic Coin

Commit:
Add Icons and Images

Commit:
Updated P2PKH and P2SH Values for MainNet

TamtamHero committed:
Display a warning when the change path of a tx is unsual

TamtamHero committed:
Add syntaxic sugar to UI

btchip committed:
Merge pull request #88 from jibeee/master

Base58 fixes

btchip committed:
Merge pull request #89 from theapplicationist/add-zclassic

Add ZClassic Coin

TamtamHero committed:
relax change path to enable non-BIP44 path

TamtamHero committed:
fix hash display in message signature on blue, fix color for dependant apps, add check on purpose/coin_type for change addresses

btchip committed:
Merge pull request #90 from TamtamHero/bip44_change_path_guard

Change path protection

btchip committed:
PR #84 Allow Ledger to sign Komodo reward claim transactions

btchip committed:
Bump version

TamtamHero committed:
fix error in change path guard

btchip committed:
Merge pull request #91 from TamtamHero/master

fix error in change path guard

btchip committed:
Bump version

TamtamHero committed:
merge UI sweetening

TamtamHero committed:
fix tx output numbering issue in Ledger Blue UI

TamtamHero committed:
updated behavior of change path warning on nano S

TamtamHero committed:
fix token display for blue, add zero padding, and remember last token approved by user

TamtamHero committed:
add macro for colors

TamtamHero committed:
whitelist P2WPKH purpose in change path guard

TamtamHero committed:
rework wording for privacy feature and change path guard

TamtamHero committed:
Nano S: add warning when displaying an address derived from an unusual path

TamtamHero committed:
Blue: add warning when displaying an address derived from an unusual path

btchip committed:
Merge pull request #92 from TamtamHero/master

Various bugfixes and quick UI refactor

btchip committed:
Bump version

btchip committed:
Merge Sapling to main branch on firmware 1.5

btchip committed:
Fix invalid CX_LAST call

btchip committed:
Merge PR #56 and bump version

btchip committed:
Review Komodo reward patch - display REWARD as fee when it's unknown rather than an inconsistent amount

btchip committed:
Fix compatibility with Blue SDK

clementperon committed:
Change png to 644

clementperon committed:
Unify macro guard headers

clementperon committed:
Add missing header license

clementperon committed:
Change Ledger Blue to Ledger App

clementperon committed:
Unify return line to Unix format

clementperon committed:
Remove trailing spaces

clementperon committed:
Remove WIDE type declaration

clementperon committed:
Remove unused variables declaration

clementperon committed:
Add prototypes declaration to fix some warnings

TamtamHero committed:
Enable PRINTF debugging

btchip committed:
Add PR #98

Commit:
Bech32 support for NIX

btchip committed:
Merge pull request #103 from NixPlatform/master

Bech32 support for NIX

btchip committed:
Merge pull request #100 from clementperon/master

Clean

btchip committed:
Merge pull request #101 from TamtamHero/master

Enable PRINTF debugging

PhilippeBonnaz committed:
Remove CX_COMPLIANCE_141 from makefile and update pending cx_ calls

btchip committed:
Bump version to release with updated U2F implementation and include WebUSB

TamtamHero committed:
Add support for Nano X

TamtamHero committed:
add settings menu and export & derivation warnings on Nano X

TamtamHero committed:
Fix crash in message signing

btchip committed:
Merge pull request #107 from PhilippeBonnaz/master

Remove CX_COMPLIANCE_141 from makefile and update pending cx_ calls

TamtamHero committed:
Merge branch 'master' into master

btchip committed:
Merge pull request #108 from TamtamHero/master

Add support for Nano X

TamtamHero committed:
remove duplicate function+ clean

TamtamHero committed:
remove Hcash and Posw coins

btchip committed:
Merge pull request #109 from TamtamHero/master

Remove deprecated coins, one duplicate function + small cleaning

TamtamHero committed:
Update with latest Nano X SDK

btchip committed:
Merge pull request #110 from TamtamHero/master

Update with latest Nano X SDK

TamtamHero committed:
fix Nano X/S dashboard icon

btchip committed:
Merge pull request #111 from TamtamHero/master

fix Nano X/S dashboard icon

btchip committed:
Bump version for fixed U2F SDK new deployment

btchip committed:
Fix Stealth transactions v2

TamtamHero committed:
Fix transaction UI on Nano X to make it more similar to the Nano S one

btchip committed:
Merge pull request #116 from TamtamHero/master

Fix transaction UI on Nano X to make it more similar to the Nano S one

TamtamHero committed:
add segwit to Qtum

btchip committed:
Merge pull request #117 from TamtamHero/master

Add segwit to Qtum

btchip committed:
Bump version

TamtamHero committed:
Enable Sapling for ZClassic Transactions

TamtamHero committed:
fix bech32 address display bug

Commit:
Bump version

btchip committed:
Merge pull request #125 from TamtamHero/master

1.3.11->1.3.13

Commit:
Add U2F heartbeat

Commit:
Make static analyzer happier

Commit:
Flowify UI

Commit:
Supports legacy SDKs

Commit:
Add missing format

Commit:
Revert to legacy USB PIDs

Commit:
Update Consensus Branch Id for Zclassic

Commit:
Bump version, remove WebUSB URL

Commit:
Update ux_flow detection

Commit:
Add LBRY

Commit:
Add Resistance

Commit:
Bump version

ryan-kengyang-alpha committed:
Merged upstream code to bring version up to 1.3.16

ryan-kengyang-alpha committed:
Added veil support fully

CaveSpectre11 commented on src/btchip_apdu_hash_input_start.c:
```diff
@@ -85,7 +85,11 @@ unsigned short btchip_apdu_hash_input_start() {
             if ((G_coin_config->kind == COIN_KIND_ZCASH) || (G_coin_config->kind == COIN_KIND_KOMODO)) {
+=======
+            if ((G_coin_config->kind == COIN_KIND_ZCASH) || (G_coin_config->kind == COIN_KIND_KOMODO) || (G_coin_config->kind == COIN_KIND_ZCLASSIC) || (G_coin_config->kind == COIN_KIND_RESISTANCE)) {
+>>>>>>> upstream/master
```
@Kryptoxic ; you have an unaddressed merge here

Kryptoxic replied:
Apologies for the late reply, I was away and busy for the month of December. I have manually looked through the entire project and merged the remaining files.

CaveSpectre11 reviewed (changes requested):
There are a number of files with unaddressed merges in it

ryan-kengyang-alpha committed:
Merged remaining files

Kryptoxic commented:
Apologies for the late reply, I was away and busy for the month of December. I have manually looked through the entire project and merged the remaining files.

ryan-kengyang-alpha committed:
Added icons

btchip committed:
Support ZCash Blossom fork

ryan-kengyang-alpha committed:
Merge remote-tracking branch 'upstream/master' into dev

Kryptoxic commented:
Updated with ledger's code from version 1.3.17

KhalilBellakrid committed:
Fix Native Segwit HRPs for DGB and LTC with respect to SLIP-173

Scotty0448 committed:
Add Ravencoin

TamtamHero committed:
Fix badly encoded icons for Lbry

btchip committed:
Merge pull request #144 from TamtamHero/fix-lbry-icons

Fix badly encoded icons for Lbry

Commit:
Bump to 1.3.19

TamtamHero committed:
Fix potential OOB write for blue app

TamtamHero committed:
Bump version 1.3.20

greenknot committed:
fix invalid hash size in btchip_public_key_hash160()

Fix false positive during fuzzing.

greenknot committed:
fix integer overflow in btchip_apdu_hash_input_finalize_full_internal()

The length passed to `cx_hash()` can be negative, leading to:
- A crash on speculos
- An error being thrown on a real device because the length is invalid

Found by libFuzzer.

TamtamHero committed:
add debugging info for tx hashing

TamtamHero committed:
Bump version 1.3.21

greenknot committed:
fix OOB-read in btchip_output_script_is_op_[create|call]

Found by libFuzzer.

greenknot committed:
refactor btchip_output_script_is_op_[create|call]

grydz committed:
Remove Legacy UI when only one output

grydz committed:
Remove btchip_output_parsing_state_t.BTCHIP_OUTPUT_HANDLE_LEGACY

grydz committed:
Bump version to 1.3.22

TamtamHero committed:
Bump version 1.3.24

TamtamHero committed:
Merge pull request #149 from TamtamHero/master

Merge back commits from Gitea

ryan-kengyang-alpha committed:
Merged upstream changes into repo

Kryptoxic commented:
Updated code with upstream changes from ledger
